Test failure on bigendian architectures

#### Minimal, reproducible code sample, a copy-pastable example if possible

```
$ python3.9 -m pytest

============================= test session starts ==============================
platform linux -- Python 3.9.2, pytest-6.0.2, py-1.10.0, pluggy-0.13.0
rootdir: /<<PKGBUILDDIR>>, configfile: pytest.ini
collected 582 items / 1 skipped / 581 selected

numcodecs/tests/test_astype.py ......                                    [  1%]
numcodecs/tests/test_base64.py ........                                  [  2%]
numcodecs/tests/test_blosc.py .......................................... [  9%]
........................................................................ [ 21%]
........................................................................ [ 34%]
.........xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx........................ [ 46%]
........................................................................ [ 59%]
..................................................                       [ 67%]
numcodecs/tests/test_bz2.py ......                                       [ 68%]
numcodecs/tests/test_categorize.py ....s.                                [ 69%]
numcodecs/tests/test_checksum32.py ......                                [ 70%]
numcodecs/tests/test_compat.py .......                                   [ 71%]
numcodecs/tests/test_delta.py ......                                     [ 73%]
numcodecs/tests/test_fixedscaleoffset.py ......                          [ 74%]
numcodecs/tests/test_gzip.py ...........                                 [ 75%]
numcodecs/tests/test_json.py .....                                       [ 76%]
numcodecs/tests/test_lz4.py .......                                      [ 78%]
numcodecs/tests/test_lzma.py ........                                    [ 79%]
numcodecs/tests/test_msgpacks.py .......                                 [ 80%]
numcodecs/tests/test_packbits.py ....                                    [ 81%]
numcodecs/tests/test_pickles.py ...s                                     [ 81%]
numcodecs/tests/test_quantize.py .......                                 [ 83%]
numcodecs/tests/test_registry.py ..                                      [ 83%]
numcodecs/tests/test_shuffle.py ........................................ [ 90%]
..................F.                                                     [ 93%]
numcodecs/tests/test_vlen_array.py .......                               [ 95%]
numcodecs/tests/test_vlen_bytes.py .......                               [ 96%]
numcodecs/tests/test_vlen_utf8.py .......                                [ 97%]
numcodecs/tests/test_zlib.py .........                                   [ 98%]
numcodecs/tests/test_zstd.py ......                                      [100%]

=================================== FAILURES ===================================
_____________________________ test_expected_result _____________________________

    def test_expected_result():
        # Each byte of the 4 byte uint64 is shuffled in such a way
        # that for an array of length 4, the last byte of the last
        # element becomes the first byte of the first element
        # therefore [0, 0, 0, 1] becomes [2**((len-1)*8), 0, 0, 0]
        # (where 8 = bits in a byte)
        arr = np.array([0, 0, 0, 1], dtype='uint64')
        codec = Shuffle(elementsize=arr.data.itemsize)
        enc = codec.encode(arr)
>       assert np.frombuffer(enc.data, arr.dtype)[0] == 2**((len(arr)-1)*8)
E       assert 0 == (2 ** ((4 - 1) * 8))
E        +  where 4 = len(array([0, 0, 0, 1], dtype=uint64))

numcodecs/tests/test_shuffle.py:144: AssertionError
=============================== warnings summary ===============================
.pybuild/cpython3_3.9_numcodecs/build/numcodecs/tests/test_json.py::test_non_numpy_inputs
.pybuild/cpython3_3.9_numcodecs/build/numcodecs/tests/test_msgpacks.py::test_non_numpy_inputs
  /usr/lib/python3/dist-packages/numpy/core/_asarray.py:83: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray
    return array(a, dtype, copy=False, order=order)

.pybuild/cpython3_3.9_numcodecs/build/numcodecs/tests/test_json.py::test_non_numpy_inputs
  /<<PKGBUILDDIR>>/.pybuild/cpython3_3.9_numcodecs/build/numcodecs/tests/test_json.py:73: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray
    assert np.array_equal(np.array(input_data), output_data)

.pybuild/cpython3_3.9_numcodecs/build/numcodecs/tests/test_msgpacks.py::test_non_numpy_inputs
  /<<PKGBUILDDIR>>/.pybuild/cpython3_3.9_numcodecs/build/numcodecs/tests/test_msgpacks.py:76: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray
    expect = np.array(input_data)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED numcodecs/tests/test_shuffle.py::test_expected_result - assert 0 == (2...
====== 1 failed, 540 passed, 3 skipped, 39 xfailed, 4 warnings in 20.96s =======
```

#### Problem description

The unit test ``numcodecs/tests/test_shuffle.py::test_expected_result`` fails on big endian architectures like
* (s390x)[https://buildd.debian.org/status/fetch.php?pkg=numcodecs&arch=s390x&ver=0.8.1%2Bds-1&stamp=1629628532&raw=0]
* (powerpc)[https://buildd.debian.org/status/fetch.php?pkg=numcodecs&arch=powerpc&ver=0.8.1%2Bds-1&stamp=1629629260&raw=0]
* (ppc64)[https://buildd.debian.org/status/fetch.php?pkg=numcodecs&arch=ppc64&ver=0.8.1%2Bds-1&stamp=1629629048&raw=0]

#### Version and installation information

Please provide the following:

* Value of ``numcodecs.__version__``: 0.8.1
* Version of Python interpreter: 3.9.2
* Operating system (Linux/Windows/Mac): GNU/Linux x86_64 (Debian Sid)
* How NumCodecs was installed (e.g., "using pip into virtual environment", or "using conda"): debian package

--- a/numcodecs/tests/test_shuffle.py
+++ b/numcodecs/tests/test_shuffle.py
@@ -133,15 +133,24 @@ def test_backwards_compatibility():
 #             codec.decode(bytearray(0))
 
 def test_expected_result():
-    # Each byte of the 4 byte uint64 is shuffled in such a way
-    # that for an array of length 4, the last byte of the last
-    # element becomes the first byte of the first element
-    # therefore [0, 0, 0, 1] becomes [2**((len-1)*8), 0, 0, 0]
-    # (where 8 = bits in a byte)
-    arr = np.array([0, 0, 0, 1], dtype='uint64')
+    # If the input is treated as a 2D byte array, with shape (size of element, number of elements),
+    # the shuffle is essentially a transpose. This can be made more apparent by using an array of
+    # big-endian integers, as below.
+    arr = np.array([0x0001020304050607, 0x08090a0b0c0d0e0f, 0x1011121314151617, 0x18191a1b1c1d1e1f],
+                   dtype='>u8')
+    expected = np.array([
+        0x00081018,
+        0x01091119,
+        0x020a121a,
+        0x030b131b,
+        0x040c141c,
+        0x050d151d,
+        0x060e161e,
+        0x070f171f,
+    ], dtype='u4')
     codec = Shuffle(elementsize=arr.data.itemsize)
     enc = codec.encode(arr)
-    assert np.frombuffer(enc.data, arr.dtype)[0] == 2**((len(arr)-1)*8)
+    np.testing.assert_array_equal(np.frombuffer(enc.data, '>u4'), expected)
 
 
 def test_incompatible_elementsize():

